### DISCUSSION TITLE
Change Corner enum order to enable bit operations

### DISCUSSION THREAD

**Discussion by lucashilles** [godotengine/godot-proposals#8152] [Engine Core]
Currently the Corner enum is ordered in such a way that it is not possible to differentiate the horizontal position using bit operations.

`core/math/math_defs.h`
``` c
enum Corner {
	CORNER_TOP_LEFT,
	CORNER_TOP_RIGHT,
	CORNER_BOTTOM_RIGHT,
	CORNER_BOTTOM_LEFT
};
```
Changing the order of the elements would make this kind of operation possible, as shown in the table.

T/B | L/R | enum
---|---|---
0 | 0 | CORNER_TOP_LEFT
0 | 1 | CORNER_TOP_RIGHT
1 | 0 | CORNER_BOTTOM_LEFT
1 | 1 | CORNER_BOTTOM_RIGHT

**Comment by AThousandShips** [comment]
This would break compatibility, so can't happen until 5.0 most likely

**Comment by Mickeon** [comment]
Sounds good, what's the necessity for it though?
